include/circt/Dialect/FIRRTL/FIRRTLExpressions.td

```diff
@@ -439,6 +443,15 @@ let inferType = "impl::inferReductionResult" in {
   }
 }
 
+def OptimizationBarrierOp : UnaryPrimOp<"optbar", FIRRTLType, FIRRTLType> {
+  let description = [{
+    This operation serves to block optimizations through it.  It may not have
+    annotations.  It has no name.  It produces no output, though may cause 
+    changes in the output to preserve the dataflow (e.g. introduce an anonymous
+    wire).
```

Comment:
"Produces no output" originally seemed incorrect to me because I was thinking that this meant "returned no result". However, you're saying that "this does not produce an operation in Verilog". It might be useful to clarify this.